src/window/mod.rs

```diff
@@ -265,12 +268,32 @@ pub fn create_window(
     let event_loop = EventLoop::new();
 
     let cmd_line_settings = SETTINGS.get::<CmdLineSettings>();
+    let frame_decoration = cmd_line_settings.frame;
+
     let winit_window_builder = window::WindowBuilder::new()
         .with_title("Neovide")
         .with_window_icon(Some(icon))
         .with_maximized(cmd_line_settings.maximized)
         .with_transparent(true)
-        .with_decorations(!cmd_line_settings.frameless);
+        // There is only two options for windows & linux, no need to match more options.
+        .with_decorations(frame_decoration == Frame::Full);
+
+    #[cfg(target_os = "macos")]
+    let winit_window_builder = match frame_decoration {
+        Frame::Full => winit_window_builder,
+        Frame::None => winit_window_builder.with_decorations(false),
+        Frame::Buttonless => window::WindowBuilder::new()
+            // This option removes the shadow & makes the margin color black(?)
+            //.with_has_shadow(false)
```

Comment:
What about this?
I don't know if shadow should be on or off, but you should either uncomment or delete this.

Reply:
I removed it.